module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -542,6 +542,42 @@ protected function fetchUserWithCql($query)
         return count($json->users) === 1 ? $json->users[0] : null;
     }
 
+    /**
+     * Helper function to retrieve paged results from FOLIO API
+     *
+     * @param string $responseKey Key containing values to collect in response
+     * @param string $interface   FOLIO api interface to call
+     * @param array  $query       CQL query
+     *
+     * @return array
+     */
+    protected function getPagedResults($responseKey, $interface, $query = [])
+    {
+        $fullResponse = [];
+        // set artificially low to test
+        $limit = 1000;
+        $offset = 0;
+
+        do {
+            $combinedQuery = compact('query', 'offset', 'limit');
+            $response = $this->makeRequest(
+                'GET',
+                $interface,
+                $combinedQuery
+            );
+            $json = json_decode($response->getBody());
```

Comment:
Right now this seems to be assuming success; are there some unhappy paths we should be accounting for? For example, would it make sense for this method to throw an exception if something goes wrong? If necessary, we could use a custom exception class that can carry the JSON response within it, so the calling code can respond to it in a more nuanced way.

Reply:
I added a pretty generic exception here to be thrown if json can't be decoded. Not sure if I understand how to set up a custom exception. I can check if the response is not a success, and throw whatever error message FOLIO gives back, and fall back to the generic error if JSON can't be decoded. What do you think?

Reply:
That sounds reasonable to me. Regarding custom exceptions, if you ever need to create one, you can just build a new class that extends `\Exception` (or one of the other standard PHP exception classes), and then you can add methods to it. This can be useful if, for example, you want the exception to be able to carry the whole API response without having to parse it out of the message. But I'm fine with just using regular `\Exception`s for now; we can always add something fancier in future when a use case arises.